x/incentives/keeper/proposals_test.go

```diff
@@ -0,0 +1,176 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	"github.com/tharsis/evmos/x/incentives/types"
+)
+
+func (suite KeeperTestSuite) TestRegisterIncentive() {
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"incentives are disabled globally",
+			func() {
+				params := types.DefaultParams()
+				params.EnableIncentives = false
+				suite.app.IncentivesKeeper.SetParams(suite.ctx, params)
+			},
+			false,
+		},
+		{
+			"inventive already registered",
+			func() {
+				regIn := types.NewIncentive(contract, allocations, epochs)
+				suite.app.IncentivesKeeper.SetIncentive(suite.ctx, regIn)
+				suite.Commit()
+			},
+			false,
+		},
+		{
+			"coin doesn't have supply",
+			func() {
+			},
+			false,
+		},
+		{
+			"allocation above allocation limit",
+			func() {
+				// Make sure the non-mint coin has supply
+				err := suite.app.BankKeeper.MintCoins(
+					suite.ctx,
+					minttypes.ModuleName,
+					sdk.Coins{sdk.NewInt64Coin(denomCoin, 1)},
+				)
+				suite.Require().NoError(err)
+
+				// decrease allocation limit
+				params := types.DefaultParams()
+				params.AllocationLimit = sdk.NewDecWithPrec(1, 2)
+				suite.app.IncentivesKeeper.SetParams(suite.ctx, params)
+			},
+			false,
+		},
+		// TODO:
```

Comment:
@fedekunze Want to check if for a given the sum of all registered allocations is > 100% , so that a new proposal can't go over this limit. Not sure whats wrong with the code or test